feat(openldap): tls certs and ldif file

guenhter opened the pull request:
Support the operations 

```
with_ldif_file(...)
with_certs(...)
```

on the OpenLdap Container

DDtKey commented on src/openldap/mod.rs:
```diff
@@ -231,6 +233,48 @@ impl OpenLDAP {
+        self
+    }
+
+    /// Set all necessary certificate artifacts to build up a secure communication.
+    /// Default: `[]`
+    pub fn with_certs(
+        mut self,
+        cert: impl Into<CopyDataSource>,
+        key: impl Into<CopyDataSource>,
+        ca: impl Into<CopyDataSource>,
+    ) -> Self {
```
probably it makes sense to split them into separate methods? (`with_cert`, `with_cert_key`, `with_cert_ca`)
Not sure about openlap requirements, but `ca` might be an optional one for example. 

However it also enables TLS, that makes sense.

WDYT of `with_tls(cert, key`) + `with_cert_ca(ca)` ? 🤔

DDtKey commented on src/openldap/create_certs.sh:
This file exists only for dev/test, right? To generate the certs. 
Let's just add a mention here in comment

Also I'd recommend to generate certs for really long period to avoid random failure in CI after a year. At least for several years

DDtKey replied:
Also, please specify `openssl` version you have used to generate the certs

guenhter replied:
Correct. The file only exists for the sake of generating the certificates. I don't mind kick the script out if you want me to. My thinking was to save someone some time in case he also needs to create them.

If you don't want me to kick it, I can also put the content of the script in a "something.md" file...

I'm ok with whatever you think is best.

DDtKey replied:
I think it's nice to have `sh` as is, but we need to document it. Either in the script itself or `readme.md` (on openldap level)

guenhter replied:
Done.

DDtKey commented on src/openldap/create_certs.sh:
```diff
@@ -0,0 +1,50 @@
+EOF
+
+# Create a self-signed root certificate
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout rootCA.key -out rootCA.crt -subj "$ROOT_SUBJECT"
+
+# Create a key for the server certificate
+openssl genrsa -out ldap.example.org.key 2048
+
+# Generate Server CSR using the config file
+openssl req -new -key ldap.example.org.key -out ldap.example.org.csr -subj "$SERVER_SUBJECT"
+
+# Sign the server certificate with the Root CA
+openssl x509 -req -in ldap.example.org.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out ldap.example.org.crt -days 365 -extfile server.cnf -extensions v3_req
```
```suggestion
openssl req -x509 -nodes -days $((365 * 20)) -newkey rsa:2048 -keyout rootCA.key -out rootCA.crt -subj "$ROOT_SUBJECT"

# Create a key for the server certificate
openssl genrsa -out ldap.example.org.key 2048

# Generate Server CSR using the config file
openssl req -new -key ldap.example.org.key -out ldap.example.org.csr -subj "$SERVER_SUBJECT"

# Sign the server certificate with the Root CA
openssl x509 -req -in ldap.example.org.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out ldap.example.org.crt -days $((365 * 20)) -extfile server.cnf -extensions v3_req
```

DDtKey commented on Cargo.toml:
```diff
@@ -111,6 +111,9 @@ clickhouse = "0.11.6"
 
+[patch.crates-io]
+testcontainers = { git = "https://github.com/guenhter/testcontainers-rs.git", branch = "release-0.23.0" }
+
```
you don't need it, we have released 0.23 already 🙂

DDtKey commented:
Thank you for the contribution! 🚀

DDtKey commented on src/openldap/mod.rs:
```diff
@@ -394,7 +444,14 @@ impl Image for OpenLDAP {
+    fn copy_to_sources(&self) -> impl IntoIterator<Item = &CopyToContainer> {
+        self.copy_to_sources
+            .iter()
+            .map(|x| x)
```
```suggestion
```

guenhter commented:
I squashed the commits to fix the "faulty" commit with '.map(|x| x)' (I honestly don't know what I was thinking there :| ) and also rebased onto the latest main

guenhter commented:
Thx for the fast response time. Appreciate that.

When do you think this will be available in a new release?

DDtKey commented:
It's already been released as [0.11.1](https://github.com/testcontainers/testcontainers-rs-modules-community/releases/tag/v0.11.1)

guenhter commented:
That is awesome. Thank you so much.
